build: use new labeling and branching in merge script

devversion opened the pull request:
We introduced a new shared configuration for merge script
labels that follow the [new proposal](https://docs.google.com/document/d/197kVillDwx-RZtSVOBtPb4BBIAw0E9RT3q3v6DZkykU).

These label semantics and the branching are set up for the Angular
framework with this commit. The goal is that labeling and merging
is consistent between all Angular projects and that clear rules
are defined for branching. This was previously not the case.

devversion committed:
build: use new labeling and branching in merge script

We introduced a new shared configuration for merge script
labels that follow the proposal of:
https://docs.google.com/document/d/197kVillDwx-RZtSVOBtPb4BBIAw0E9RT3q3v6DZkykU

These label semantics and the branching are set up for the Angular
framework with this commit. The goal is that labeling and merging
is consistent between all Angular projects and that clear rules
are defined for branching. This was previously not the case.

jelbourn reviewed (approved):
LGTM - we should coordinate merging this with the docs updates, any other tooling required (like ngbot), and actually renaming the labels

gkalpak commented on .ng-dev/merge.ts:
```diff
@@ -1,38 +1,25 @@
  * are respected by the merge script (e.g. the target labels).
  */
-export const merge = (): MergeConfig => {
-  // TODO: resume dynamically determining patch branch
```
Don't we need this comment in `requiredBaseCommits['10.0.x']` now? :grin:

devversion replied:
Not quite sure. I think it would be better to set the required base commits for branches explicitly. Consider the scenario where we set a required base commit for `10.0.x` and then suddenly we have a new patch branch for `10.1.x`. The required
base commit set for `10.0.x` is not necessarily part of `10.1.x` that has been based on `master`.

petebacondarwin replied:
In that case I would imagine having a separate file (called `patch-version.ts`??) that only contains the current patch branch, if we are not computing it, rather than hiding it in here. The trouble with explicit is that we then have to have processes to update it during significant releases.

devversion committed:
ci: update github robot to reflect new target labels

Updates the Github robot to reflect the updated target
labels that are used as part of the canonical versioning
and labeling for the Angular organization.

josephperrott reviewed (approved):
LGTM
